src/main/java/com/google/devtools/build/lib/windows/WindowsSubprocess.java

```diff
@@ -140,35 +141,32 @@ public Thread newThread(Runnable runnable) {
     stderrStream =
         stderrRedirected ? null : new ProcessInputStream(WindowsProcesses.getStderr(nativeProcess));
     stdinStream = new ProcessOutputStream();
-    waitLatch = new CountDownLatch(1);
     // Every Windows process we start consumes a thread here. This is suboptimal, but seems to be
     // the sanest way to reconcile WaitForMultipleObjects() and Java-style interruption.
-    @SuppressWarnings("unused")
-    Future<?> possiblyIgnoredError = WAITER_POOL.submit(this::waiterThreadFunc);
+    processFuture = WAITER_POOL.submit(this::waiterThreadFunc);
   }
 
-  private void waiterThreadFunc() {
+  // Waits for the process to finish. Returns 'false' if the process exited
+  // normally, 'true' in case of a timeout and throws an IllegalStateException
+  // when stuff goes really wrong.
+  private boolean waiterThreadFunc() {
     switch (WindowsProcesses.waitFor(nativeProcess, timeoutMillis)) {
       case 0:
         // Excellent, process finished in time.
-        break;
+        return false;
 
       case 1:
-        // Timeout. Terminate the process if we can.
-        timedout.set(true);
-        WindowsProcesses.terminate(nativeProcess);
```

Comment:
Sorry - this line technically belongs into the last commit where I add the IOPort stuff (because that then handles the termination in case of timeout, so we no longer have to do it here). :)